requirements.in

```diff
@@ -1,26 +1,65 @@
-# normally pulled in by kombu, but version 2.2.0 is broken
-amqp<=2.1.4
-# also handles Celery as a requirement
-selinon[celery]
-sqlalchemy
-psycopg2
-lxml
-beautifulsoup4
-# We install a patched version from dnf as we cannot use requests from PyPI - we need own certificates
-#requests
```

Comment:
Can you please audit whether all of these direct dependencies are really used? I suspect no.

Reply:
sqlalchemy, lxml are being used. not sure about amqp but its used for rabbit mq. Please verify this before removing them

Reply:
Yes, I have checked and removed unused packages.

Reply:
I just tried with a tool called `pipreqs` which infers the dependency list by scanning the source code, it list only the following direct dependencies,

```
requests==2.24.0
pip==19.3.1
anymarkup==0.8.1
beautifulsoup4==4.9.1
boto3==1.14.34
botocore==1.17.34
celery==4.4.7
flexmock==0.10.4
git2json==0.2.3
GitPython==3.1.7
jsl==0.2.4
jsonschema==3.2.0
lxml==4.5.2
pytest==6.0.1
PyYAML==5.3.1
raven==6.10.0
requests_futures==1.0.0
selinon==1.1.0
semantic_version==2.8.5
SQLAlchemy==1.3.18
Werkzeug==1.0.1
```

Can you just try the above list without any version pinning(into requirements.in)? Of course you need to add https://github.com/fabric8-analytics/fabric8-analytics-utils and https://git@github.com/fabric8-analytics/fabric8-analytics-version-comparator.git.

Reply:
There is no usage of https://git@github.com/fabric8-analytics/fabric8-analytics-version-comparator.git in the worker.

Reply:
@yzainee if fabric8-analytics-version-comparator is a dependency of utils, then pip will automatically pull it right? Why should we add it here if it is not directly used by workers?

Reply:
Actually its not doing so. Never got a chance to investigate why. Probably a good time to check that out

Reply:
@yzainee Fixed the transitive pull issue with the external packages. https://github.com/fabric8-analytics/fabric8-analytics-utils/pull/58

Reply:
+1

Reply:
@jparsai Please make use of the latest utils. Also remove all unwanted dependencies using `pipreqs`.